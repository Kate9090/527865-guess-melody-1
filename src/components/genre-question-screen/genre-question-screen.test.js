import React from 'react';
import renderer from 'react-test-renderer';

import GenreQuestionScreen from './genre-question-screen.jsx';

const mock = {
  genre: `rock`,
  answers: [
    {
      genre: `rock`,
      src: ``,
    },
    {
      genre: `blues`,
      src: ``,
    },
    {
      genre: `jazz`,
      src: ``,
    },
    {
      genre: `jazz`,
      src: ``,
    },
  ]
};

it(`GenreQuestionScreen renders correctly`, () => {
  const {
    genre: genre,
    answers: answers,
  } = mock;
  const tree = renderer
    .create(<GenreQuestionScreen
      genre = {genre}
      answers = {answers}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

