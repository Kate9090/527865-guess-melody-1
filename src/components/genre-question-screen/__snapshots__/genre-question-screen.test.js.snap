// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionScreen renders correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    rock
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id={0}
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="rock"
        />
        <label
          className="game__check"
          htmlFor={0}
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id={1}
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="pop"
        />
        <label
          className="game__check"
          htmlFor={1}
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id={2}
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="jazz"
        />
        <label
          className="game__check"
          htmlFor={2}
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id={3}
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="blues"
        />
        <label
          className="game__check"
          htmlFor={3}
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
